closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -1001,15 +1036,16 @@ public synchronized int doUpdate(byte[] in, int inOff, int inLen, byte[] out,
         @Override
         public synchronized int doFinal(byte[] in, int inOff, int inLen, byte[] out,
                 int outOff) throws ShortBufferException, KeyException {
+            if ((in == null) && (inLen != 0)) {
+                throw new ShortBufferException("Input buffer is null, while input length is non-zero.");
+            }
```

Comment:
Why is this check done only for `EngineAEADEnc` (and not also for `EngineAEADDec` and `EngineStreamOnly`)?

Reply:
For `EngineAEADEnc`, if the user provides a null `in`, but a non-zero `inLen`, a number of bytes will be left empty in the output array before writing the tag. There will be no indication that this has happened or an exception of any kind. This is why I added the check.

As far as `EngineStreamOnly` goes, it doesn't really matter since the value of `inLen` is never used if `in == null`. The only issue could be with the return statement in the `else` scenario which I could change to zero.

Lastly, in `EngineAEADDec`, there is a scenario where an issue would occur if `doUpdate()` is never called and `doFinal()` is called with `(in == null) && (inLen != 0)`. In that case, the native method will return -1 and an exception will be thrown, even though it might not be very helpful indeed.

I can, for completeness, add the same check in the other two engines, if you feel that would be the right thing to do.

Reply:
Looking at `CipherSpi` it seems nulls are not expected in that API. Perhaps this checking belongs at that level instead of in the classes that implement `ChaChaEngine`?

Reply:
Input can be null, in case you just want to finalize without passing any further information. But you're right, this check would be more suited in the `engineDoFinal()` of `NativeChaCha20Cipher`. I will move it there.

Reply:
As you mentioned, the check for `(in == null) && (inLen != 0)` is not required, as it is a scenario that cannot occur. All those possibilities are checked as part of the calls to the methods of `Cipher`.

I have removed the check altogether.